comp/core/remoteagentregistry/impl/remoteagentregistry.go

```diff
@@ -497,10 +568,16 @@ func newRemoteAgentClient(registration *remoteagentregistry.RegistrationData) (p
 }
 
 type remoteAgentDetails struct {
-	lastSeen    time.Time
-	displayName string
-	apiEndpoint string
-	client      pb.RemoteAgentClient
+	lastSeen     time.Time
+	displayName  string
+	apiEndpoint  string
+	client       pb.RemoteAgentClient
+	configStream *configStream
```

Comment:
Does it need to be a pointer? If not we can remove some nil checks we are doing in the code. As all remoteAgentdetails have a dedicated configStream 😄

Reply:
updated and removed the now unnecessary nil checks 🙂